File: lib/java/src/transport/TServerTransport.java
// Copyright (c) 2006- Facebook
// Distributed under the Thrift Software License
//
// See accompanying file LICENSE or visit the Thrift site at:
// http://developers.facebook.com/thrift/

package com.facebook.thrift.transport;

/**
 * Server transport. Object which provides client transports.
 *
 * @author Mark Slee <mcslee@facebook.com>
 */
public abstract class TServerTransport {

  public abstract void listen() throws TTransportException;

  public final TTransport accept() throws TTransportException {
    TTransport transport = acceptImpl();
    if (transport == null) {
      throw new TTransportException("accept() may not return NULL");
    }
    return transport;
  }

  public abstract void close();

  protected abstract TTransport acceptImpl() throws TTransportException;
}

File: lib/csharp/src/Transport/TServerTransport.cs
//
//  TServerTransport.cs
//
//  Begin:  Dec 3, 2007
//  Authors:
//		Will Palmeri <wpalmeri@imeem.com>
//
//  Copyright (C) 2007 imeem, inc. <http://www.imeem.com>
//  All rights reserved.
using System;

namespace Thrift.Transport
{
	public abstract class TServerTransport
	{
		public abstract void Listen();
		public abstract void Close();
		protected abstract TTransport AcceptImpl();

		public TTransport Accept()
		{
			TTransport transport = AcceptImpl();
			if (transport == null) {
			  throw new TTransportException("accept() may not return NULL");
			}
			return transport;
		 }
	}
}
